raiden/network/transport/matrix/utils.py

```diff
@@ -197,7 +197,7 @@ def refresh_address_presence(self, address: Address) -> None:
         should **not** generally be used.
         """
         composite_presence = {
-            self._fetch_user_presence(uid) for uid in self._address_to_userids[address]
+            self._fetch_user_presence(uid) for uid in self.get_userids_for_address(address)
```

Comment:
I am not 100% sure whether this resolves the issue. 

`self.get_userids_for_address` would return the `set` of user IDs for a particular address... which practically points to the same set that `self._address_to_userids[address]` would return. Think of them as both pointers to the same object. If that's the case, then increasing the size of the set somewhere else would cause the same error.

I would suggest making a copy of the set using `.copy()` or converting it to a list `list(self._address_to_userids[address])`